content/blog/minecraft-data-packs-scoreboards/index.md

```diff
@@ -0,0 +1,348 @@
+
+```shell
+#                     set our scoreboard entry
+#                     |   use the entry of the current player
+#                     |   |                    use "1" as the value to add
+#                     |   |                    |
+scoreboard objectives set @s fennifith.animals_spawned 1
+```
+
+<div style="margin-top: -2em;">
+
+| Target    | Value |
+| --------- | ----- |
+| fennifith | 1     |
+
+</div>
+
+If we want to add to this value, we can use the `scoreboard objectives add` subcommand instead. Likewise, `scoreboard objectives remove` will subtract a value from our scoreboard.
+
+```shell
+#                     add a number to the current scoreboard value
+#                     |   use the entry of the current player
+#                     |   |                    use "2" as the number to add
+#                     |   |                    |
+scoreboard objectives add @s fennifith.animals_spawned 2
+```
+<div style="margin-top: -2em;">
+
+| Target    | Value |
+| --------- | ----- |
+| fennifith | 3     |
+
+</div>
+
+### Using global entries
+
+While these entries allow us to store player-specific numbers, we might also want a value that affects our entire data pack. For example, we might want to track the total number of animals spawned in our world in addition to the number of animals for each player.
+
+We can do this by referencing a *nonexistent player*. The scoreboard will include an entry for any entity or name, regardless of whether it actually exists in our world - so by using an invalid name as the target, we can reference it globally from anywhere in our code.
+
+```shell
+scoreboard objectives set $global fennifith.animals_spawned 4
+```
+
+<div style="margin-top: -2em;">
+
+| Target    | Value |
+| --------- | ----- |
+| fennifith | 3     |
+| $global   | 4     |
+
+</div>
+
+If we didn't include the `$` before our variable name in this snippet, our code would still work! However, what would happen if a player registered the username `global` and tried to use our data pack?
+
+Since the `$` is an invalid username character, we can safely use it for global values without that possibility.
+
+### Using the "/execute store" subcommand
+
+Each Minecraft command provides a "success" and a "result" value which specify if the command was successful - and if so, what value it returned.
+
+The `execute store` subcommand can be used to designate a place to store these values, such as a scoreboard entry.
+
+For example, this command will copy the value of our `$global` variable into `$global_2`...
+
+```shell
+#       store the result of the command
+#       |                  place it in "$global_2"
+#       |                  |                                       run a command that returns the value of "$global"
+#       |                  |                                       |
+execute store result score $global_2 fennifith.animals_spawned run scoreboard players get $global fennifith.animals_spawned
```

Comment:
Storing values needs to be better represented, to show that an entry can have multiple values for multiple objectives